src/core/ddsi/src/ddsi_typebuilder.c

```diff
@@ -1081,7 +1081,7 @@ static dds_return_t get_ops_struct (const struct typebuilder_struct *tb_struct,
     uint32_t flags = 0u;
     flags |= tb_struct->members[m].is_external ? DDS_OP_FLAG_EXT : 0u;
     flags |= tb_struct->members[m].is_optional ? (DDS_OP_FLAG_OPT | DDS_OP_FLAG_EXT) : 0u;
-    flags |= (tb_struct->members[m].is_key || parent_is_key) ? DDS_OP_FLAG_KEY : 0u;
```

Comment:
This surprises me: I'm pretty sure the serializer tracks this "parent is key" thing internally because of cases like `A::x` in:
```
struct A { long x; };
struct B { @key A u; A v; };
```
and so I doubt the "parent is key" here did anything useful. Still, it is a surprising change that must've been prompted by something that I am not seeing just yet.